File: output/ChatGPT/avatar/Python/C++/codeforces_412_A.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int n, k;
    cin >> n >> k;
    string v;
    cin >> v;
    vector<string> d = {"LEFT", "RIGHT"};
    bool f = k - 1 < n - k;
    int m = min(k - 1, n - k);
    vector<string> a;
    for (int i = 0; i < m; i++) {
        a.push_back(d[!f]);
    }
    for (int i = v.size() - 1 + 2 * f; i >= 0 && i < v.size(); i += 2 * f - 1) {
        a.push_back("PRINT " + v[i]);
        a.push_back(d[f]);
    }
    for (int i = 0; i < a.size() - 1; i++) {
        cout << a[i] << endl;
    }


    return 0;
}

File: output/ChatGPT/avatar/Python/Java/codeforces_412_A.java
import java.util.Scanner;
import java.util.ArrayList;

public class codeforces_412_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int k = scanner.nextInt();
        scanner.nextLine();
        String v = scanner.nextLine();

        String[] d = {"LEFT", "RIGHT"};
        boolean f = k - 1 < n - k;
        int m = Math.min(k - 1, n - k);
        ArrayList<String> a = new ArrayList<>();

        for (int i = 0; i < m; i++) {
            a.add(d[f ? 1 : 0]);
        }

        for (int i = v.length() - 1; i >= 0 && f; i -= 2) {
            a.add("PRINT " + v.charAt(i));
            a.add(d[f ? 0 : 1]);
        }

        for (int i = 0; i < v.length() && !f; i += 2) {
            a.add("PRINT " + v.charAt(i));
            a.add(d[f ? 0 : 1]);
        }

        for (int i = 0; i < a.size() - 1; i++) {
            System.out.println(a.get(i));
        }
    }
}